### ISSUE TITLE
EPIC: Convert to typescript React

### ISSUE THREAD

**Issue by trickydisco78** [trickydisco78/global-opps-ts#1]
Convert codebase to typescript React
* Set up tslint, tsconfig
* Add interfaces for state and props
* type declearations for functions and variables
* Models folder for data
* Split components into folders : index.tsx, interfaces, css modules

**Comment by mbicknese** [comment]
Looking at this issue and the things you want to change. Would it be an idea to start from scratch and rebuild the application "the right way"? 

https://github.com/Microsoft/TypeScript-React-Starter should get us started pretty quickly.

**Comment by trickydisco78** [comment]
Yes, this is what i'm thinking too. It's too much of a rewrite. Is it best to create a new branch off of this or create a new repo and move issues (if you can do that?)

**Comment by trickydisco78** [comment]
You learn something everyday. Didn't know about orphaned branches
